tests/caasp/first_boot.pm

```diff
@@ -20,7 +20,11 @@ use caasp 'process_reboot';
 
 sub run {
     # On VMX images bootloader_uefi eats grub2 needle
-    assert_screen 'grub2' unless is_caasp('VMX');
+    if (is_caasp 'DVD') {
+        if (!check_screen('grub2')) {
```

Comment:
As you probably know I recommend to always prevent the use of `check_screen` with non-zero timeout. Also, a `record_soft_failure` should normally be only used for product issue tracking. If you do not need this check and do not even abort the waiting time within the grub bootloader why not just delete the whole section and just wait in the next step for `linux-login-casp`? What Panos mentioned in https://progress.opensuse.org/issues/28648#note-6 is that he does not want to change the autoyast profile to disable the grub timeout. Maybe you want to rethink this?

Reply:
So if I understood it right, the `installation/autoyast.pm` is going to check also for the `linux-login-casp` needle among others. That's great because this needle is the final stage which means the system *has been installed* **and** *booted*. Well, in case this happens, it means that `first_boot` has already been tested by the `installation/autoyast.pm` which makes itself a bit useless, but I am fine with it. I guess this is why you need to exclude the `grub` needle from the `first_boot.pm`, because it's *too* late for that. This makes sense, and it should work in case openQA gets stalled again.

PS: As for the `record_soft_failure` I would agree with Oliver. Please feel free to use `record_info('poo#28648', "Stall prevented matching of BIOS needle");`.

Verification run for maintenance: http://skyrim.qam.suse.de/tests/overview?build=4652-kravciak&version=2.0&distri=caasp&groupid=98

Reply:
Fixed